src/Tools/dotnet-trace/CommandLine/Commands/ReportCommand.cs

```diff
@@ -0,0 +1,162 @@
+                        {
+                            CallTreeNodeBase node = callTreeNodes[index];
+                            index++;
+                            if(!unwantedMethodNames.Any(node.Name.Contains))
+                            {
+                                nodesToReport.Add(node);
+                                count++;
+                            }
+                        }
+
+                    PrintReportHelper.TopNWriteToStdOut(nodesToReport, inclusive, verbose);
+                }
+                return await Task.FromResult(0);
+            }
+            catch(Exception ex)
+            {
+                Console.Error.WriteLine($"[ERROR] {ex.ToString()}");
+            }
+
+            return await Task.FromResult(0);
+        }
+
+        public static Command ReportCommand() =>
+            new Command(
+                name: "report",
+                description: "Generates a report into stdout from a previously generated trace.")
+                {
+                    //Handler
+                    HandlerDescriptor.FromDelegate((ReportDelegate)Report).GetCommandHandler(),
+                    //Options
+                    FileNameArgument(),
+                    new Command(
+                        name: "topN",
+                        description: "Finds the top N methods that have been on the callstack the longest.")
+                        {
+                            //Handler
+                            HandlerDescriptor.FromDelegate((TopNReportDelegate)TopNReport).GetCommandHandler(),
+                            TopNOption(),
+                            InclusiveOption(),
+                            VerboseOption(),
+                        }
+                };
+
+        private static Argument<string> FileNameArgument() =>
+            new Argument<string>("trace_filename")
+            {
+                Name = "tracefile",
+                Description = "The file path for the trace being analyzed.",
+                Arity = new ArgumentArity(1, 1)
+            };
+
+        private static int DefaultN() => 5;
+
+        private static Option TopNOption()
+        {
+            return new Option(
+                aliases: new[] {"-n", "--number" },
+                description: $"Gives the top N methods on the callstack.")
+                {
+                    Argument = new Argument<int>(name: "n", getDefaultValue: DefaultN)
+                };
+        }         
+
+        private static bool DefaultIsInclusive => false;
+
+        private static Option InclusiveOption() =>
+            new Option(
+                aliases: new[] { "--inclusive" },
+                description: $"Output the top N methods based on inclusive time. If not specified, exclusive time is used by default.")
+                {
+                    Argument = new Argument<bool>(name: "inclusive", getDefaultValue: () => DefaultIsInclusive)
```

Comment:
Do we plan to have the defaults configurable at any point? It seems unnecessarily complex to pass in a function as the default value, I assume the defaults will never change so getting rid of DefaultIsInclusive, DefaultN, etc and replacing with `getDefaultValue: () => false`, etc would be easier to read